index.html

```diff
@@ -2766,134 +2728,114 @@ <h3>Consumption</h3>
-    "signatureValue": "o9r6LxgoGN8FoaeeUA6EdDcv12GvDzFEmCgjWzvpur2YSQyA8W2r0SSWUK+nH5tMqzaFLun6wwZ1Eot37amGDg==",
-    "type": "ed25519Signature2018"
-  },
-  "verificationMethod": [
-    {
-      "curve": "ed25519",
-      "expires": "2019-12-01T03:00:00Z",
-      "id": "bafyreicubtx5wqo3nosc4cazrkctfhwd6rewezgpwoe4swirls4ebdhs2i;example:key=idbafyreicubtx5wqo3nosc4cazrkctfhwd6rewezgpwoe4swirls4ebdhs2i",
-      "publicKeyBase64": "qmz7tpLNKKKdl7cD7PbejDiBVp7ONpmZbfmc7cEK9mg=",
-      "type": "EdDsaPublicKey"
-    }
-  ],
-  "updated": "2020-05-01T03:00:00Z"
-}
+<pre class="example nohighlight" title="DID Document as DagCBOR with tag 42 used in Proof section">
+  a5                                      # map(5)
+  62                                   # text(2)
+     6964                              # "id"
+  78 40                                # text(64)
+     6469643a6578616d706c653a313244334b6f6f574d4864727a6377706a6264725a7335474771455241766367715833623564707550745061396f743639796577 # "did:example:12D3KooWMHdrzcwpjbdrZs5GGqERAvcgqX3b5dpuPtPa9ot69yew"
+  65                                   # text(5)
+     70726f6f66                        # "proof"
+  d8 2a                                # tag(42)
+     58 26                             # bytes(38)
+        000190011b20a7f3a13d0dafa9b8b640fee52173c27b9a66ff27f6669b3be73fc0ee6849e2ce # "\x00\x01\x90\x01\e \xA7\xF3\xA1=\r\xAF\xA9\xB8\xB6@\xFE\xE5!s\xC2{\x9Af\xFF'\xF6f\x9B;\xE7?\xC0\xEEhI\xE2\xCE"
+  68                                   # text(8)
+     40636f6e74657874                  # "@context"
+  78 1c                                # text(28)
+     68747470733a2f2f7777772e77332e6f72672f6e732f6469642f7631 # "https://www.w3.org/ns/did/v1"
+  6e                                   # text(14)
+     61757468656e7469636174696f6e      # "authentication"
+  81                                   # array(1)
+     78 83                             # text(131)
+        6469643a6578616d706c653a313244334b6f6f574d4864727a6377706a6264725a7335474771455241766367715833623564707550745061396f7436397965773b6b65792d69643d6261667972656963756274783577716f336e6f73633463617a726b63746668776436726577657a6770776f6534737769726c733465626468733269 # "did:example:12D3KooWMHdrzcwpjbdrZs5GGqERAvcgqX3b5dpuPtPa9ot69yew;key-id=bafyreicubtx5wqo3nosc4cazrkctfhwd6rewezgpwoe4swirls4ebdhs2i"
+  72                                   # text(18)
+     766572696669636174696f6e4d6574686f64 # "verificationMethod"
+  81                                   # array(1)
+     78 40                             # text(64)
+        6469643a6578616d706c653a313244334b6f6f574d4864727a6377706a6264725a7335474771455241766367715833623564707550745061396f743639796577 # "did:example:12D3KooWMHdrzcwpjbdrZs5GGqERAvcgqX3b5dpuPtPa9ot69yew"
           </pre>
         </section>
+      </section>
+    </section>
+    <section>
+      <h3>Converting Data between Core Representations</h3>
```

Comment:
I think this section may be problematic.. Even though CBOR-to-JSON conversion may be a special case where optimizations are possible, "converting" between representations should always go via the abstract data model (ADM). I.e. to "convert" from CBOR to JSON, you would first run the CBOR consumption process, and then the JSON production process. The end result should be the same anyway.

Reply:
Since the core data model is defined by https://w3c.github.io/did-core/#data-model (essentially 2 sentences)... I find this paragraph extremely helpful.

If we just replace the word "JSON" with "INFRA" I think most of this still works.... I would prefer to see this section retained, with changes to make it between CBOR and INFRA, and I would argue that the other representations need a similar section or should be removed.

Suggest changing the section name to "Abstract Data Model Conversion" or similar.

@jonnycrunch I assume CBOR -> ADM -> JSON will retain all "properties", you may want to add an explicit sentence about that here.

Reply:
If it's changed to say "CBOR to INFRA" instead of "CBOR to JSON", then shouldn't that simply be the "Consumption" section (which right now seems almost empty in this PR)?

Reply:
@OR13 ,  it would be cleaner to use CDDL as the ADM, in which case this is simple.  I am new to writing spec-text so I may have glossed over the details and would be grateful for any help.  personally, I find the definition of an Abstract Data Model, ... well too abstract, which is why I like the `Data Definition Language` that CDDL provides much more precise and something I can wrap my head around and do testing, while it does provide great support for explicit Type definition, Classes and well-defined required and optional Properties across representations. so CBOR <--> CDDL <--JSON --> and even YAML.

Reply:
@jonnycrunch I completely agree, would prefer to use CDDL instead of english and handwaving at INFRA.

Reply:
-1 to CDDL -- it is a concrete schema language, not an abstract data model. I'm concerned about the confusion around this... if we want to use CDDL to define everything, then we need to get rid of the abstract data model (which is technically possible... but it would require a fairly large rewrite of the specification at this point).

Reply:
@msporny CDDL is a normative Data Definition Language and is more than just a schema check for concrete representations.  CDDL provides a machine-processable notation to express CBOR (and JSON) data structures. It  is used in many specifications and other design documents for describing the structures that one party is meant to send to another, enabling informed discussion, input validation, and even code generation. It also works for both JSON (including JSON-LD) and CBOR.  I would like everyone to seriously consider CDDL. 

My understanding regarding what we are attempting to solve with the Abstract Data Model is class of objects whose logical behavior is defined by a set of values and a set of operations.  

CDDL allows us to unambiguously define classes of object with logical behaviors and works with JSON and CBOR.  CDDL in combination with sets of operations defined in the Core Spec achieves this goal.   

I would also invite @cabo to weigh in his thoughts and appreciate any advice he may have.  As I mentioned, I was anticipating feedback from Jim Schaad (@jimsch) who unfortunately died last month.

Reply:
We are pretty rich in data modeling and poor in information modeling formal languages.
Sometimes a data modeling language, with some understanding of its limitations, can stand in for an information modeling language.
I haven't analyzed this document, so I don't have an opinion yet.  
But I do know that CDDL has helped in other cases.
(There are converters from CDDL to the formats proposed on json-schema.org, see how we made use of automatic conversion in https://www.ietf.org/archive/id/draft-onedm-t2trg-sdf-00.html#name-json-schemaorg-rendition-of -- we had to model CDDL features by generating two versions of the json-schema.org format.)

Reply:
Whether or not CDDL should replace INFRA as the language of the ADM is a question that needs to be resolved by the working group. Until such time as the working group decides to use CDDL in the ADM, I agree with @OR13 that replacing "JSON" with "INFRA" is the appropriate thing to do here. And I agree with @peacekeeper that a description of CBOR -> INFRA belongs in the consumption section.

Reply:
Perhaps this section could be as simple as
"Conversion from the CBOR representation to another supported representation is a process of using the CBOR consumption rules defined here, followed by the using the production rules for the chosen representation. Conversely, conversion from another representation to CBOR is a process of following that representation's consumption rules, followed by using the CBOR production rules defined here."

Reply:
I think that it would be helpful to flush out the Abstract Data Model and I'd like to make my case that CDDl as a Data Definition Language can stand in for the data model aspects of the ADM as I have demoed before.  Happy to present next week for the TPAC sessions next week to make my argument.  @cabo, as I have discussed with the Chair and will their approval, I'd like to extend an invitation to be an external Invited Expert and help use through this dilemma.   Despite being invited, there is a formal [process](https://www.w3.org/participate/invited-experts/) to accept and join our working group.

Reply:
I agree that the conversion section should be removed from here.

Reply:
Agreed this this section doesn't make sense given the direction of the specification. We have an ADM for a reason and general rules for production/consumption as well. The Editors will be cleaning these sections up in future PRs.